dhis-2/dhis-support/dhis-support-system/src/test/java/org/hisp/dhis/system/util/ValidationUtilsTest.java

```diff
@@ -186,50 +212,313 @@ void testUsernameIsValid()
     void testDataValueIsZeroAndInsignificant()
     {
         DataElement de = new DataElement( "DEA" );
-        de.setValueType( ValueType.INTEGER );
+
+        de.setValueType( INTEGER );
         de.setAggregationType( AggregationType.SUM );
         assertTrue( dataValueIsZeroAndInsignificant( "0", de ) );
+
         de.setAggregationType( AggregationType.AVERAGE_SUM_ORG_UNIT );
         assertFalse( dataValueIsZeroAndInsignificant( "0", de ) );
     }
 
     @Test
-    void testDataValueIsValid()
+    void testValueIsComparable()
+    {
+        testIntegerTypes();
+        testDoubleTypes();
+        testBooleanTypes();
+        testDateTimeTypes();
+        testStringTypes();
+    }
+
+    private void testIntegerTypes()
```

Comment:
I would say put test annotations on each method, makes it easier to identify which test fails.

Reply:
Done @larshelge